File: pset1/cash/cash.c
#include<stdio.h>
#include<math.h>
#include<cs50.h>
#include<ctype.h>

int main(void)
{
    int rem, coins, temp ;
    float change ;

    do
    {
        change = get_float("Enter the Change :");
    }
    while (change < 0);

    rem = round(change * 100);

    //printf(" %d", rem);

    coins = round(rem / 25);
    rem -= coins * 25;

    while (rem > 0)
    {
        if (rem >= 10)
        {
            temp = round(rem / 10);
            rem -= (temp * 10);
        }

        else if (rem >= 5 && rem < 10)
        {
            temp = round(rem / 5);
            rem -= (temp * 5);
        }

        else
        {
            temp = rem / 1;
            rem -= (temp * 1);
        }

        coins += temp;
    }

    printf(" %i\n", coins) ;
}

File: pset5/mario/cash.py
from cs50 import get_float

while True:
    cents = get_float("Enter the change:")
    if cents > 0:
        break

cents = (int)(cents * 100)
coins = 0

if cents > 25:
    coins = cents // 25
    cents = cents - coins * 25

if cents >= 10:
    coinsT = cents // 10
    cents -= coinsT * 10
    coins += coinsT

if cents >= 5:
    coinsF = cents // 5
    cents -= coinsF * 5
    coins += coinsF

coins += cents

print(coins)
